### ISSUE TITLE
Deleting the default theme breaks everything

### ISSUE THREAD

**Issue by Ace2213** [jacobwb/hashover-next#275]
Since I'm going to use the dark theme, I figured I wouldn't need the others so I deleted them, which caused the entire thing to stop working.

I had not looked into that directory and didn't know it had files that continue to be relevant even when using other themes.

Either those files should be included in other themes or something in the docs should mention that the default theme contains core elements.

**Comment by jacobwb** [comment]
I don't know if this is the best way to describe it, but the current design of the theming system is a file override. Allowing every theme to use their own CSS and HTML files, while at the same time allowing specific files to be omitted from the theme in favor of the defaults. In this way, the default theme serves as a fallback for any file that doesn't exist in all other themes.

#102 suggests moving the `images` directory into the default theme. There is an argument that the opposite should happen, rename `images` to `assets` and move the core files in the default theme into this directory, allowing all themes to use these default files or provide their own.

Both arguments are valid, but I've not given it much thought.

At the same time, I wouldn't recommend deleting the default theme. Doing so would be like going into the classes directory and deleting classes you don't intend to use. With specific settings some classes aren't used, but deleting them may still cause problems. So in general, I would suggest not deleting any files included with the project if you want it to work as intended.

Let me know which approach you think is more logical, and I will take it into consideration when I eventually try to make the overall file structure more consistent.

**Comment by Ace2213** [comment]
Makes sense. Usually solutions that work this way have the default theme files hidden within the core files though, to avoid issues like this. Having them as a theme option gives the impression that they're optional.

This way, in the absence of any themes, the default theme is used.

Another possible solution that requires a lot less work is simply adding a disclaimer either in the install docs or near the theme selector in the settings page explaining that the default theme contains core files and shouldn't be touched.

**Comment by jacobwb** [comment]
Yes. I am leaning towards the `assets` directory approach, this would mean the default theme files would be "hidden within the core files" as you say.

I am working on a complete rewrite of how the interface is created. The rewrite would mean removing three classes, `FormUI`, `CommentsUI` and `HTMLTag`, in favor of one -- much smaller -- `UI` class. The rewrite isn't done, but it would likely reduce the code for the interface by 50% at least while allowing complete control over the HTML structure, meaning the comment form, the container elements, and everything else in the UI would be easily adjustable.

When I push this rewrite, I will also change the file structure to move the default theme files out of the themes directory to avoid giving the impression that these files are optional.
